### PR TITLE
Consent Screens UI - BiometrcConsent, Terms&Conditions Consent, OtherConsents

### PR DESCRIPTION
**Pull request by Princy2909**
### Biometric Consent Screen
 It displays a screen with an image, header text, explanatory text, and a button. The component uses state to toggle between light and dark mode. When the button is pressed, it logs the message "Permission Granted" to the console. The strings used in the component are extracted as constants at the beginning of the file for easier translation or modification.

The component includes the following elements:
A top image
Header text
Explanatory text with a padlock icon
Warning text
A button with an arrow icon and text
The styles for the component are defined in a StyleSheet object, which allows for efficient rendering of the component.
**TESTING EVIDENCE**

for light mode:-
Screenshot_1718867511

for Dark Mode :-
Screenshot_1718867527

### CODE REVIEW

**Comment by Princy2909** [comment]
Terms&Conditions Screen and OtherConsents Screen UI added.

**Testing Evidence for OtherConsents Screen**
Screenshot_1718948109
Screenshot_1718948115

**Testing Evidence for Terms & Conditions Screen**

Screenshot_1718949967
Screenshot_1718949975

File: BiometricConsent.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            {/* Button with arrow icon and text */}
+            <TouchableOpacity style={styles.button} onPress={handleButtonPress}>
+                <Image source={require('./assets/arrow.png')} style={styles.arrow} />
+                <Text style={styles.buttontext}>{BUTTON_TEXT}</Text>
+            </TouchableOpacity>
+        </View>
+    );
+};
+
+const styles = StyleSheet.create({
```
Comment (Jha2508) [inline BiometricConsent.tsx]: move styles out

File: OtherConsents.tsx [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+                            <Text style={styles.okButtonText}>{OK_BTN_TEXT}</Text>
+                        </TouchableOpacity>
+                    </View>
+                </View>
+            </Modal>
+        </View>
+    );
+};
+
+const styles = StyleSheet.create({
```
Comment (Jha2508) [inline OtherConsents.tsx]: move styles out

File: OtherConsents.tsx [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+            <Text style={styles.header}>{LOCATION_HEADER}</Text>
+            <Text style={styles.text}>{LOCATION_TEXT}</Text>
+        </View>
+    );
+};
+
+/**
+ * NotificationsConsent component
+ */
+const NotificationsConsent = () => {
```
Comment (Jha2508) [inline OtherConsents.tsx]: move to a separate file

File: OtherConsents.tsx [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+const OK_BTN_TEXT = 'OK';
+const NEXT_BTN_TEXT = 'Next';
+const AGREEMENT_TEXT = 'I agree to the terms and conditions';
+const PERMISSION_HEADER = 'We need your permission';
+const PERMISSION_PARAGRAPH = 'To have a smooth account creation process with no hiccups and safeguard your account';
+
+/**
+ * LocationConsent component
+ */
+const LocationConsent = () => {
```
Comment (Jha2508) [inline OtherConsents.tsx]: move to a separate file

File: OtherConsents.tsx [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+const LOCATION_HEADER = 'Location permission';
+const NOTIFICATION_TEXT = 'We need your permission to send you notifications.';
+const LOCATION_TEXT = 'We need your location to provide you with a better experience.';
+const PERMISSION_REQUIRED_HEADER = 'Permission Required';
+const PERMISSION_ALERT_MESSAGE = 'Please check the checkbox to grant permission.';
+const OK_BTN_TEXT = 'OK';
+const NEXT_BTN_TEXT = 'Next';
+const AGREEMENT_TEXT = 'I agree to the terms and conditions';
+const PERMISSION_HEADER = 'We need your permission';
+const PERMISSION_PARAGRAPH = 'To have a smooth account creation process with no hiccups and safeguard your account';
```
Comment (Jha2508) [inline OtherConsents.tsx]: move all these to a constants file

File: TermsConditions.tsx [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+const TERMS_BTN = 'KUBAIR Terms';
+const SECURE_BTN = 'KUBAIR Secure';
+const TITLE = 'Understand our Terms & Conditions';
+const SUBTITLE = 'Written in a way humans can understand & relate';
+const AGREEMENT = 'I agree to the above terms & conditions that I am Indian citizen, my country of tax residence is India and I am not (nor am related to) a politically exposed person';
+const DATA_SHARE = 'I agree to share personal data with KUBAIR and partners, to be accessed, stored & used for providing services';
+const NEXT_BTN = 'Next';
+const ALERT_TITLE = 'Permission Required';
+const ALERT_MSG = 'Please check the checkbox to grant permission.';
+const OK_BTN = 'OK';
```
Comment (Jha2508) [inline TermsConditions.tsx]: move constants to a different file

File: TermsConditions.tsx [hunk 7]
```diff
@@ -0,0 +1,303 @@
...
+                            <Text style={styles.okButtonText}>{OK_BTN}</Text>
+                        </TouchableOpacity>
+                    </View>
+                </View>
+            </Modal>
+        </View>
+    );
+};
+
+const styles = StyleSheet.create({
```
Comment (Jha2508) [inline TermsConditions.tsx]: move styles out

File: BiometricConsent.tsx (hunk 1 above)
Comment (Princy2909) [reply BiometricConsent.tsx]: Done

File: OtherConsents.tsx (hunk 2 above)
Comment (Princy2909) [reply OtherConsents.tsx]: Done

File: OtherConsents.tsx (hunk 3 above)
Comment (Princy2909) [reply OtherConsents.tsx]: Done

File: OtherConsents.tsx (hunk 4 above)
Comment (Princy2909) [reply OtherConsents.tsx]: Done

File: OtherConsents.tsx (hunk 5 above)
Comment (Princy2909) [reply OtherConsents.tsx]: Done

File: TermsConditions.tsx (hunk 6 above)
Comment (Princy2909) [reply TermsConditions.tsx]: Done

File: TermsConditions.tsx (hunk 7 above)
Comment (Princy2909) [reply TermsConditions.tsx]: Done

File: src/UI/Components/ConsentScreenComp/OtherConsentComponents.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+import { View, Text, Image} from 'react-native';
+import styles from '../../Screens/ConsentScreens/Style/OtherConsentStyles';
+import { OtherConsentsConstants } from '../../Screens/ConsentScreens/Constants';
+
+const constants = OtherConsentsConstants();
+
+/**
+ * LocationConsent component
+ */
+const LocationConsent = () => {
```
Comment (Jha2508) [inline src/UI/Components/ConsentScreenComp/OtherConsentComponents.tsx]: Why two components for the same purpose ? make them one and take parameters in props

File: src/UI/Components/ConsentScreenComp/TermsConditionsComponents.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+import { TermsConditionsConstants } from '../../Screens/ConsentScreens/Constants';
+import { config } from '../../Screens/ConsentScreens/Config';
+
+const constants = TermsConditionsConstants();
+const { TERMS_URL, SECURE_URL} = config;
+
+/**
+ * Terms component
+ */
+const Terms = () => {
```
Comment (Jha2508) [inline src/UI/Components/ConsentScreenComp/TermsConditionsComponents.tsx]: same as above

Comment (Princy2909) [reply src/UI/Components/ConsentScreenComp/TermsConditionsComponents.tsx]: Done

File: src/UI/Components/ConsentScreenComp/OtherConsentComponents.tsx (hunk 8 above)
Comment (Princy2909) [reply src/UI/Components/ConsentScreenComp/OtherConsentComponents.tsx]: Done

**Review by Jha2508 — APPROVED** [review]

**Pull request merged**